re-enable fabric in 2-erisc mode on BH as experimental

SeanNijjar opened the pull request:
This conditionally enables 2-erisc mode for TT-Fabric for improved performance. The enabled mode adopts a simple, interim parallelization scheme where erisc0 services sender channels and erisc1 services receiver channels.

Observed up to 60% link utilization with these changes, @ 4k packet size on BH QB.

To run with this mode, enable `TT_METAL_FABRIC_BLACKHOLE_TWO_ERISC=1 TT_METAL_MULTI_AERISC=1`.

To enable this mode, multiple Ethernet TXQs must be programmed. This has an issue when re-initializing fabric as a different topology. This is likely due to a bug with the txq/rxq reprogramming sequence and some messages being directed to the incorrect rxq on fabric reinits.

### Checklist
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI: https://github.com/tenstorrent/tt-metal/actions/runs/18549461977
- [x] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable): https://github.com/tenstorrent/tt-metal/actions/runs/18549463143

aagarwalTT commented on tests/tt_metal/tt_metal/perf_microbenchmark/routing/test_fabric_test_2erisc_quick_at_least_2x2_mesh.yaml:
```diff
@@ -0,0 +1,35 @@
+Tests:
```
also the file name says atleast 2x2, is that intended?

aagarwalTT commented on .github/workflows/blackhole-multi-card-unit-tests-impl.yaml:
```diff
@@ -71,6 +71,7 @@ jobs:
               TT_METAL_CLEAR_L1=1 ./build/test/tt_metal/perf_microbenchmark/routing/test_tt_fabric --test_config ${TT_METAL_HOME}/tests/tt_metal/tt_metal/perf_microbenchmark/routing/test_fabric_sanity_common.yaml
               TT_METAL_CLEAR_L1=1 ./build/test/tt_metal/perf_microbenchmark/routing/test_tt_fabric --test_config ${TT_METAL_HOME}/tests/tt_metal/tt_metal/perf_microbenchmark/routing/test_fabric_sanity_at_least_2x2_mesh.yaml
               TT_FABRIC_PROFILE_RX_CH_FWD=1 TT_METAL_CLEAR_L1=1 ./build/test/tt_metal/perf_microbenchmark/routing/test_tt_fabric --test_config ${TT_METAL_HOME}/tests/tt_metal/tt_metal/perf_microbenchmark/routing/test_fabric_code_profiling.yaml
+              TT_METAL_FABRIC_BLACKHOLE_TWO_ERISC=1 TT_METAL_MULTI_AERISC=1 TT_METAL_CLEAR_L1=1 ./build/test/tt_metal/perf_microbenchmark/routing/test_tt_fabric --test_config ${TT_METAL_HOME}/tests/tt_metal/tt_metal/perf_microbenchmark/routing/test_fabric_test_2erisc_quick_at_least_2x2_mesh.yaml
```
last I remember these multi-card pipelines run on a variety of systems (and 1X2) as well. If we want to only run on atleast 2X2 we may need some extra skipping logic

SeanNijjar replied:
I'm going to remove it (see my comment above)

SeanNijjar commented on tests/tt_metal/tt_metal/perf_microbenchmark/routing/test_fabric_test_2erisc_quick.yaml:
```diff
@@ -0,0 +1,35 @@
+Tests:
```
Not for the time being. I'm actually going to update the PR to remove the invocation from pipelines because it seems to be unstable (see my original comment in the PR description).

SeanNijjar committed:
conditionally enable 2-erisc fabric mode

To use this mode, run with:
`TT_METAL_FABRIC_BLACKHOLE_TWO_ERISC=1 TT_METAL_MULTI_AERISC=1`

Limitations: You can only run one fabric type (1D, ring, mesh) at a time
and running on a new topology will require a reset.
